Code/GraphMol/Chirality.cpp

```diff
@@ -1112,7 +1113,15 @@ void findPotentialStereoBonds(ROMol &mol, bool cleanIt) {
             // ------------------
             // get the CIP ranking of each atom if we need it:
             if (!cipDone) {
-              Chirality::assignAtomCIPRanks(mol, ranks);
+              if (!begAtom->hasProp(common_properties::_CIPRank)) {
+                Chirality::assignAtomCIPRanks(mol, ranks);
+              } else {
+                // no need to recompute if we don't need to recompute. :-)
+                for (unsigned int ai = 0; ai < mol.getNumAtoms(); ++ai) {
```

Comment:
Surprisingly works ok?  Here is how I thought I could trick it.

```
 {  // simple
    std::string smiles = "CC=CC";
    ROMol *m = SmilesToMol(smiles);
    TEST_ASSERT(m);
    TEST_ASSERT(m->getNumAtoms() == 4);
    TEST_ASSERT(m->getBondWithIdx(1)->getBondType() == Bond::DOUBLE);
    TEST_ASSERT(m->getBondWithIdx(1)->getStereo() == Bond::STEREONONE);

    MolOps::findPotentialStereoBonds(*m, true);
    TEST_ASSERT(m->getBondWithIdx(1)->getStereo() == Bond::STEREOANY);

    // should be nothing to do here...
    ROMol *mh = MolOps::addHs(*m);
    MolOps::findPotentialStereoBonds(*mh, false);
    std::cout << "num atoms " << mh->getNumAtoms() << std::endl;
    TEST_ASSERT(mh->getNumAtoms() == 12);
    TEST_ASSERT(m->getBondWithIdx(1)->getStereo() == Bond::STEREOANY);

    // should raise a key error
    MolOps::findPotentialStereoBonds(*mh, true);
    
    delete m;
    delete mh;
  }
```

Now I admit to being more confused....  I would like a test that catches the key error but I haven't been able to make one, I think we can move on though.